clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -85,42 +90,47 @@
     private static final Schema FETCH_RESPONSE_V0 = new Schema(
             new Field(RESPONSES_KEY_NAME, new ArrayOf(FETCH_RESPONSE_TOPIC_V0)));
 
+    // V1 bumped for the addition of the throttle time
     private static final Schema FETCH_RESPONSE_V1 = new Schema(
             THROTTLE_TIME_MS,
             new Field(RESPONSES_KEY_NAME, new ArrayOf(FETCH_RESPONSE_TOPIC_V0)));
-    // Even though fetch response v2 has the same protocol as v1, the record set in the response is different. In v1,
-    // record set only includes messages of v0 (magic byte 0). In v2, record set can include messages of v0 and v1
-    // (magic byte 0 and 1). For details, see Records, RecordBatch and Record.
+
+    // V2 bumped to indicate the client support message format V1 which uses relative offset and has timestamp.
     private static final Schema FETCH_RESPONSE_V2 = FETCH_RESPONSE_V1;
 
-    // The partition ordering is now relevant - partitions will be processed in order they appear in request.
+    // V3 bumped for addition of top-levl max_bytes field and to indicate that partition ordering is relevant
     private static final Schema FETCH_RESPONSE_V3 = FETCH_RESPONSE_V2;
 
-    // The v4 Fetch Response adds features for transactional consumption (the aborted transaction list and the
+    // V4 adds features for transactional consumption (the aborted transaction list and the
     // last stable offset). It also exposes messages with magic v2 (along with older formats).
-    private static final Schema FETCH_RESPONSE_ABORTED_TRANSACTION_V4 = new Schema(
-            new Field(PRODUCER_ID_KEY_NAME, INT64, "The producer id associated with the aborted transactions"),
-            new Field(FIRST_OFFSET_KEY_NAME, INT64, "The first offset in the aborted transaction"));
+    // aborted transaction field names
+    private static final Field.Int64 LAST_STABLE_OFFSET = new Field.Int64("last_stable_offset",
```

Comment:
I agree it's subjective. I thought it was useful to initialize the field along with the schema version that introduced it. I think it's a bit nasty no matter what to tell you the truth. I hope eventually we're not hand-coding these schemas anymore.

Reply:
I am not very strong on this. On the other hand I don't feel strong about committing this refactor change based on one person's opinion. @ijuma Could you comment which style is preferred?

Reply:
Yeah, I don't have a strong preference either. The main thing I was trying to accomplish is avoiding the duplication of the type and the documentation string. Where we put the field is less of a concern.

Reply:
So it looks like there is no clearly better choice. Typically I feel it is probably better to refactor code if the new solution is better. The reason is mainly to reduce the code conflict with patches from other contributors and also to reduce the need for other contributors to learn the new pattern.

Since @ijuma seems OK with the new style, we can do the refactor as is.